docs/micro-services/index.md

```diff
@@ -0,0 +1,82 @@
+
+Clone git repository.
+
+```bash
+git clone https://github.com/powsybl/powsybl-spring.git
+```
+
+Compile and build docker images.
+
+```bash
+mvn -Pdocker install
+```
+
+Docker images should be available in your local registry.
+
+```bash
+$ docker images
+powsybl/powsybl-server-discovery                 1.0.0-SNAPSHOT                   b4aa78912a87        49 years ago        163.7 MB
+powsybl/powsybl-server-discovery                 latest                           b4aa78912a87        49 years ago        163.7 MB
+powsybl/powsybl-server-storage                   1.0.0-SNAPSHOT                   5d5dfdc64985        49 years ago        170.9 MB
+powsybl/powsybl-server-storage                   latest                           5d5dfdc64985        49 years ago        170.9 MB
+powsybl/powsybl-server-config                    1.0.0-SNAPSHOT                   33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-config                    latest                           33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-network                   1.0.0-SNAPSHOT                   c6d5501e3af9        49 years ago        188.1 MB
+powsybl/powsybl-server-network                   latest                           c6d5501e3af9        49 years ago        188.1 MB
+```
+
+Four services have been built:
+ - Config service which is responsible for centralizing configuration of all others services. It is implemented using [Spring Cloud Config](https://spring.io/projects/spring-cloud-config).
+ - Discovery service which is responsible for service registry. It is implemented using [Netflix Eureka](https://github.com/Netflix/eureka).
+ - Storage service, is a low level web service responsible for file system like data storage.
+ - Network service rely on Storage service and allow network data query.
+
+## Services configuration
+
+### AFS configuration
+
+Create `$HOME/.itools/config.yml` and add following configuration to add MapDB storage named 'test':
+
+```yaml
+mapdb-app-file-system:
+        drive-name: test
+        db-file: /storage/db
+        remotely-accessible: true
+```
+
+This file is be shared with server storage container.
+
+Create `$HOME/storage` directory. This directory is also be shared with server storage container and contains MapDB data.
+
+### Spring config service configuration
+
+Create `$HOME/config` directory and copy all Spring Boot service configuration files.
+
+```bash
+mkdir $HOME/config
+cp powsybl-spring/config/* $HOME/config/
+```
+
+## Starting containers
+
+To start docker containers using docker compose:
+
+```bash
+docker-compose up
+```
+
+## Getting Swagger documentation
+
+Storage and Network services are documented using [Swagger](https://swagger.io/) and are available once container have been started. 
+ - [Storage service documentation](http://192.168.0.12:8090/swagger-ui.html)
+ - [Network service documentation](http://192.168.0.13:8091/swagger-ui.html)
```

Comment:
These links are not working whereas the containers are up:
```
$> docker container ls
CONTAINER ID        IMAGE                                     COMMAND                  CREATED             STATUS                   PORTS                    NAMES
9e4c54cf3a5b        powsybl/powsybl-server-network:latest     "java -cp /app/resou…"   2 minutes ago       Up 2 minutes (healthy)   0.0.0.0:8091->8091/tcp   powsybl-spring_network-server_1
5976357add29        powsybl/powsybl-server-storage:latest     "java -cp /app/resou…"   2 minutes ago       Up 2 minutes (healthy)   0.0.0.0:8090->8090/tcp   powsybl-spring_storage-server_1
c04e7ca69a92        powsybl/powsybl-server-discovery:latest   "java -cp /app/resou…"   3 minutes ago       Up 3 minutes (healthy)   0.0.0.0:8061->8061/tcp   powsybl-spring_discovery-server_1
3e2c406d6cf6        powsybl/powsybl-server-config:latest      "java -cp /app/resou…"   3 minutes ago       Up 3 minutes (healthy)   0.0.0.0:8088->8088/tcp   powsybl-spring_config-server_
```

The configuration seems OK:
```
$> docker port powsybl-spring_storage-server_1
8090/tcp -> 0.0.0.0:8090

$> docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' powsybl-spring_storage-server_1
192.168.0.12
```

Reply:
The connection works properly if I'm offline (cf. remark on the corresponding [PR](https://github.com/powsybl/powsybl-spring/pull/8))